addons/source-python/packages/source-python/core/__init__.py

```diff
@@ -101,6 +107,36 @@ def _unload_instance(self):
             'have its own implementation of an _unload_instance method.')
 
 
+class WeakAutoUnload(AutoUnload):
+    """Subclass of AutoUnload used to unload specific instances when all
+    references to the instance are deleted.
+    """
+
+    def new(cls, *args, **kwargs):
+        self = super().__new__(cls)
+        self.___unloaded = False
```

Comment:
Yeah, thanks, it seems this should have been 2 underscores as per https://www.python.org/dev/peps/pep-0008/ (to avoid naming collisions for subclasses).